src/Kunstmaan/PagePartBundle/Event/PagePartEvent.php

```diff
@@ -18,9 +19,12 @@ final class PagePartEvent extends Event
      */
     private $response;
 
-    public function __construct(PagePartInterface $pagePart)
+    private ?HasPagePartsInterface $page;
+
+    public function __construct(PagePartInterface $pagePart, ?HasPagePartsInterface $page = null)
     {
         $this->pagePart = $pagePart;
+        $this->page = $page;
```

Comment:
Can you add a deprecation here if $page is null, so we can make it madatory in the next major as it is logic to have the page always available. Also add an entry to the upgrade file that the $page will be required in 8.0

Reply:
@acrobat  Done